sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1214,6 +1215,13 @@ class CastSuite extends CastSuiteBase {
       checkEvaluation(Cast(Literal(134.12), DecimalType(3, 2)), null)
     }
   }
+
+  test("SPARK-30008: collect_list/collect_set can cast to ArrayType not containsNull") {
```

Comment:
Since nullability is just a hint for optimization, this is an improvement, isn't it?

Reply:
yea, I'll remove the prefix